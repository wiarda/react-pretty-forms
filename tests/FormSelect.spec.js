/* global describe test expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { FormSelect } from '../index';

describe('Select component', () => {
  test('Matches snapshot', () => {
    const component = renderer
      .create(<FormSelect name="email" />, {
        createNodeMock: element => {
          if (element.type === 'select') {
            return {
              current: {
                value: 'test value',
              },
            };
          }
        },
      })
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders required props', () => {
    const component = renderer
      .create(<FormSelect name="test" label="This is a test field" />, {
        createNodeMock: element => {
          if (element.type === 'select') {
            return {
              current: {
                value: 'test value',
              },
            };
          }
        },
      })
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders required field', () => {
    const component = renderer
      .create(<FormSelect name="test" label="This is a test field" required />, {
        createNodeMock: element => {
          if (element.type === 'select') {
            return {
              current: {
                value: 'test value',
              },
            };
          }
        },
      })
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Renders without a validation message', () => {
    const component = renderer
      .create(
        <FormSelect name="test" label="This is a test field" validationMessage={false} required />,
        {
          createNodeMock: element => {
            if (element.type === 'select') {
              return {
                current: {
                  value: 'test value',
                },
              };
            }
          },
        },
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Accepts custom styles', () => {
    const component = renderer
      .create(
        <FormSelect
          name="test"
          label="This is a test field"
          className="test-class"
          validationMessage={false}
          required
        />,
        {
          createNodeMock: element => {
            if (element.type === 'select') {
              return {
                current: {
                  value: 'test value',
                },
              };
            }
          },
        },
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Accepts option children', () => {
    const component = renderer
      .create(
        <FormSelect
          name="test"
          label="This is a test field"
          className="test-class"
          validationMessage={false}
          required
        >
          <option value="">-</option>
          <option value="Dr.">Dr.</option>
          <option value="Prof.">Prof.</option>
        </FormSelect>,
        {
          createNodeMock: element => {
            if (element.type === 'select') {
              return {
                current: {
                  value: 'test value',
                },
              };
            }
          },
        },
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Invalid initialValues render as defaultValue', () => {
    const componentInitial = renderer
      .create(
        <FormSelect
          name="test"
          label="This is a test field"
          className="test-class"
          validationMessage={false}
          initialValue="uhoh"
          required
        >
          <option value="">-</option>
          <option value="Dr.">Dr.</option>
          <option value="Prof.">Prof.</option>
        </FormSelect>,
        {
          createNodeMock: element => {
            if (element.type === 'select') {
              return {
                current: {
                  value: 'test value',
                },
              };
            }
          },
        },
      )
      .toJSON();

    const componentDefault = renderer
      .create(
        <FormSelect
          name="test"
          label="This is a test field"
          className="test-class"
          validationMessage={false}
          initialValue="uhoh"
          required
        >
          <option value="">-</option>
          <option value="Dr.">Dr.</option>
          <option value="Prof.">Prof.</option>
        </FormSelect>,
        {
          createNodeMock: element => {
            if (element.type === 'select') {
              return {
                current: {
                  value: 'test value',
                },
              };
            }
          },
        },
      )
      .toJSON();

    expect(componentInitial).toMatchSnapshot();
    expect(componentDefault).toMatchSnapshot();
  });
});
