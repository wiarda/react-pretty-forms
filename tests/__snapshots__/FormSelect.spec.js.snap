// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Accepts custom styles 1`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="test-class"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={true}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Select component Accepts option children 1`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="test-class"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={true}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
      <option
        value=""
      >
        -
      </option>
      <option
        value="Dr."
      >
        Dr.
      </option>
      <option
        value="Prof."
      >
        Prof.
      </option>
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Select component Invalid initialValues render as defaultValue 1`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="test-class"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={true}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
      <option
        value=""
      >
        -
      </option>
      <option
        value="Dr."
      >
        Dr.
      </option>
      <option
        value="Prof."
      >
        Prof.
      </option>
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Select component Invalid initialValues render as defaultValue 2`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="test-class"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={true}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
      <option
        value=""
      >
        -
      </option>
      <option
        value="Dr."
      >
        Dr.
      </option>
      <option
        value="Prof."
      >
        Prof.
      </option>
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Select component Matches snapshot 1`] = `
<div
  className="pretty-row"
>
  <select
    className="form--pretty-select"
    defaultValue="unselected"
    multiple={false}
    name="email"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    required={false}
    size={1}
  >
    <option
      disabled={true}
      hidden={true}
      value="unselected"
    />
  </select>
  ;
  <div
    className="pretty-validation"
    data-validity={true}
  >
    Please enter a valid email address
  </div>
</div>
`;

exports[`Select component Renders required field 1`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="form--pretty-select"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={true}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    Please fill in this required field
  </div>
</div>
`;

exports[`Select component Renders required props 1`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="form--pretty-select"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={false}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;

exports[`Select component Renders without a validation message 1`] = `
<div
  className="pretty-row"
>
  <label
    className="pretty-label"
    data-active={true}
    data-validity={true}
    htmlFor="test"
  >
    <span
      className="pretty-label-text"
      data-active={true}
    >
      This is a test field
    </span>
    <select
      className="form--pretty-select"
      defaultValue="unselected"
      multiple={false}
      name="test"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      required={true}
      size={1}
    >
      <option
        disabled={true}
        hidden={true}
        value="unselected"
      />
    </select>
  </label>
  <div
    className="pretty-validation"
    data-validity={true}
  >
    
  </div>
</div>
`;
